### ISSUE TITLE
Inferred type of "self" can be assigned to any type in Callable

### ISSUE THREAD

**Issue by AndrewLester** [microsoft/pyright#821]
**Describe the bug**
The type that is given to "self" arguments in methods can be assigned to any type in a Callable.

**To Reproduce**
Create a method (with a "self" argument) and that method is then assignable to a Callable with any type as its first argument. The example below uses a Callable with its first argument set to a float, but any type will work.
```python
from typing import Callable

class TestClass:
    def method(self) -> None:
        pass
    
func1: Callable[[float], None] = TestClass.method
```

**Expected behavior**
An error should appear on the line for `func1`.

**Screenshots or Code**
![Screen Shot 2020-07-10 at 11 25 51 PM](https://user-images.githubusercontent.com/23221268/87215463-b7d9a100-c304-11ea-83e6-b83d0ff4b104.png)

**VS Code extension or command-line**
VS Code extension version: v1.1.51

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in Pyright 1.1.52, which I just published. It will be included in the next published version of Pylance.
